department-of-veterans-affairs/caseflow
# Correspondence Cases: Action Required: ReturnedPackageTask

Resolves [Jira Issue Title](https://jira.devops.va.gov/browse/APPEALS-54936)

# Description
Action type column has been updated to Returned for the Return To Inbound Ops task and When a user clicks the link they are taken to the Correspondence Details page for the Correspondence.

## Acceptance Criteria

## Testing Plan

1. Log in as an inbound ops team admin user
2. Navigate to /queue/correspondence/team and click on a correspondence
3. Under General Information, select "other motionsnd save changes.
4. Scroll down to the bottom and select "Create record".
5. Continue through step 1, and scroll to "tasks not related to appeal" on step 2.
6.  Click "+ Add tasks", select an other motion related task in the drop down, and add text in the textbox.
7.  Click "Continue" from the second step. Before continuing, copy the link from the address bar to use later!
8. On step 3, click the "submit" button. Then Confirm the submission.
9. Now that the data is set up, you can continue to test the story.

1. Log in as a LIT_SUPPORT_USER
2. Paste the address saved from the previous step, make sure to remove the last part of /review_package or /intake
3. This will take you to the Correspondence details page for that appeal 
4. Click on the + button next to the "Tasks not related to appeal" section
5. In the Actions drop down next to the other motion task, verify the "Return to Inbound Ops Team" task is present
6. Click on the return to inbound ops, and the modal will appear.
7. The return button should initially not be clickable until a radio button is selected.
8. Once a radio button is selected, the return button will be clickable
9. If "Other" is selected, the return button will remain locked until you type a reason for return.
10. Click on the "Return" button. 
11. Now  an inbound ops team admin user
12. Navigate to /queue/correspondence/team 
13. And check for the Veteran we have created the data , see that the Action type will be "Returned"
14. Click on the Veteran details link user should navigate to Correspondence details page.


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
